### PR TITLE
Implemented ApiKey REST interface

### PR DESCRIPTION
**Pull request by KerstenBreuer**
First draft of the REST interface with ApiKey handling.

### CODE REVIEW

**Comment by KerstenBreuer** [comment]
@lkuchenb, any idea why the `datameta/api/openapi.yaml` is not found in the testing setup?

**Comment by lkuchenb** [comment]
Probably not installed with the python package, that's something that's easy to miss when you develop with `-e`. It would need to go into the `MANIFEST.in` file then.

**Comment by KerstenBreuer** [comment]
Great, now it works, thanks @lkuchenb

**Comment by KerstenBreuer** [comment]
By the way, you can see the swagger UI at http://localhost:8080/api 😄

**Review by MoritzHahn1337 — APPROVED** [review]
As far as my limited knowledge goes, this looks good.

**Comment by KerstenBreuer** [comment]
@lkuchenb, I implemented the changes we discuss so that the apikey post can also be used when relying on cookies.

Could you please have another look?

**Comment by KerstenBreuer** [comment]
Following changes:
- Validation errors are now forwarded as correct response
- the application/json is now required but the properties (emails and passwords) are not when used from a cookie-base loggedin browser (you can simply provide an empty json as request)

@lkuchenb , @MoritzHahn1337 to try it out (or implement) in the frontend, something like this will work:
```
fetch(
    'http://localhost:8080/api/keys', 
    {
        method: "POST",
        body: JSON.stringify({}),
        headers: {
        "Content-type": "application/json; charset=UTF-8"
   }
})
   .then(res => res.json())
   .then(console.log)
```

I hope this is fine.

**Review by MoritzHahn1337 — APPROVED** [review]

File: datameta/api/apikeys.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            }
+
+def generate_api_key(request:Request, user:User):
+    token = "".join(choice(ascii_letters+digits) for _ in range(64) )
+
+    db = request.dbsession
+    apikey = ApiKey(
+        user_id = user.id,
+        value = token,
+        comment = "",
```
Comment (lkuchenb) [inline datameta/api/apikeys.py:59]: This should be implemented

**Review by lkuchenb — CHANGES_REQUESTED** [review]

File: datameta/api/apikeys.py (hunk 1 above)
Comment (KerstenBreuer) [reply datameta/api/apikeys.py:59]: @lkuchenb, what do you mean by that?

**Review by lkuchenb — APPROVED** [review]

**Pull request merged**